drivers/usb/device/usb_dc_rpi_pico.c

```diff
@@ -635,15 +635,27 @@ int usb_dc_ep_read_continue(uint8_t ep)
 		return -EINVAL;
 	}
 
-	bytes_received = state.setup_available ?
-			 sizeof(struct usb_setup_packet) :
-			 udc_rpi_get_ep_buffer_len(ep);
+	bool data_packet_after_setup_is_expected = false;
+	if (state.setup_available) {
+		const struct usb_setup_packet *setup = (const void *)&usb_dpram->setup_packet;
+		if ((setup->wLength != 0U) ||
+		    !usb_reqtype_is_to_host(setup)) {
```

Comment:
I think it would be clearer if you used `usb_reqtype_is_to_device(setup)` instead.

Reply:
Will do, well, this particular line will disappear, but will double-check to not use !X() if instead the equivalent Y() is available without being negated

Reply:
Done, no more usage of !usb_reqtype_is_to_host(